controller/fsm.py

```diff
@@ -0,0 +1,109 @@
+"""Module that implements `drunc` finite state machine."""
+
+from statemachine import State, StateMachine
+
+from . import controller_interface as ci
+
+
+class DruncFSM(StateMachine):
+    """Finite state machine for the `drunc` controller."""
+
+    # The states the controller can be in
+    none = State("None", initial=True)
+    initial = State("Initial")
+    configured = State("Configured")
+    ready = State("Ready")
+    running = State("Running")
+    dataflow_drained = State("Dataflow_Drained")
+    triggered_sources_stopped = State("Triggered_Sources_Stopped")
+
+    # The transitions between the states
+    boot = none.to(initial)
+    terminate = initial.to(none)
+    conf = initial.to(configured)
+    scrap = configured.to(initial)
+    start = configured.to(ready)
+    enable_triggers = ready.to(running)
+    disable_triggers = running.to(ready)
+    drained_dataflow = ready.to(dataflow_drained)
```

Comment:
Should this maybe be:
```suggestion
    drain_dataflow = ready.to(dataflow_drained)
```